### PR TITLE
Issue #244: Create cgmanifest.json for the project

### PR DESCRIPTION
**Pull request by shaahji**
Issue #244: Create cgmanifest.json for the project

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+          "Name": "torch",
+          "Version": "1.11.0"
+        }
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "aaa5eab410c2e3fe41ebf7a2316be9c51572dbd2",
```
Comment (wenbingl) [inline cgmanifest.json:151]: comes from any dependency?

**Comment by wenbingl** [comment]
How do you generate this file? can we have some script to do it?

File: cgmanifest.json [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+          "Name": "onnxruntime",
+          "Version": "1.11.1"
+        }
+      }
+    },
+    {
+      "component": {
+        "type": "pip",
+        "pip": {
+          "Name": "onnxconverter",
```
[the remark this answers is not in the document]
Comment (shaahji) [reply cgmanifest.json:97]: I have had to install this to get build working.

File: cgmanifest.json [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+          "repositoryUrl": "https://gitlab.kitware.com/cmake/cmake"
+        }
+      }
+    },
+    {
+      "component": {
+        "type": "pip",
+        "pip": {
+          "Name": "cmake",
+          "Version": "3.22.5"
```
[the remark this answers is not in the document]
Comment (shaahji) [reply cgmanifest.json:161]: ONNXRuntime lists it in their version so I presume it's required.

File: cgmanifest.json (hunk 1 above)
Comment (shaahji) [reply cgmanifest.json:151]: It's version 3.18.2.

File: cgmanifest.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (wenbingl) [reply cgmanifest.json:97]: I search pypi.org, and there is no onnxconverter, where do you install it from?

**Comment by shaahji** [comment]
> It is not about what are necessary for building this repo from source locally, it is about what are used in ort-extension packages. Especially, what are used for building the packages that your team publishes.

Wouldn't those two be the same, at least for the most part. Everything in that file is needed to successfully build locally and should be needed in CI as well. There might be some version differences depending on which version of compiler you use or what version of python are you running. This list might still be _incomplete_ since there might be more dependencies needed for steps that we don't necessarily run locally. Call it minimal list of dependencies.

File: cgmanifest.json [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+          "commitHash": "v3.7.3",
+          "repositoryUrl": "https://github.com/nlohmann/json.git"
+        }
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "4.5.4",
```
Comment (snnn) [inline cgmanifest.json:52]: opencv has a lot of dependencies. For example, libpng. If it was used when you build your package, then you need to list it here as well.  Transitively. 
Without the dependencies, opencv itself doesn't do any useful thing.

Comment (shaahji) [reply cgmanifest.json:52]: Okay! I will cover the transitive dependencies as well. How accurate should be versions be? When the requirement is a range which should be included, the lower or upper?

Comment (snnn) [reply cgmanifest.json:52]: Should be exactly the version that you use for building your packages. Exactly. Not a range.

**Comment by snnn** [comment]
For example, if your package has OCOS_ENABLE_OPENCV_CODECS=ON, which means your binary contains jpeg code. But jpeg has patents. You will need to pay someone to get the authorization first. If you don't put jpeg in the cgmanifest.json, I don't know you are using jpeg. Then I wouldn't let know you how to start the legal process. Then it's pirating. We should carefully avoid it happening. It's like, when you write a paper, you must be careful on the references. If you didn't do it right, it may ruin your academic life.

**Comment by wenbingl** [comment]
> For example, if your package has OCOS_ENABLE_OPENCV_CODECS=ON, which means your binary contains jpeg code. But jpeg has patents. You will need to pay someone to get the authorization first. If you didn't put jpeg in the cgmanifest.json, then I wouldn't know you are using jpeg, then I wouldn't ask to you to start the legal process. Then it may become pirating. Therefore we should be careful on such things. It's like, when you write a paper, you must carefully list all the references. If you didn't do it right, it may ruin your academic life. So, a "minimal list“ is not enough. The list should be full and complete and accurate.

jpeg patent expires, so the CODEC should be same as another library as well.

**Comment by snnn** [comment]
> jpeg patent expires, so the CODEC should be same as another library as well.

There are two things: patents, and license agreement. Even patents are expired, we still need to register all such things in cgmanifest.json then put them in ThirdPartyNotice.txt file.

**Review by snnn — APPROVED** [review]

**Comment by snnn** [comment]
@wenbingl , please take a look.

**Review by wenbingl — APPROVED** [review]

**Pull request merged**